private/bufpkg/bufpolicy/policy_config.go

```diff
@@ -27,3 +36,204 @@ type PolicyConfig interface {
 	// PluginConfigs returns the PluginConfigs for the File.
 	PluginConfigs() []bufconfig.PluginConfig
 }
+
+// MarshalPolicyConfigAsJSON marshals the PolicyConfig to a stable JSON representation.
+//
+// This is used for the O1 digest and should not be used for other purposes.
+// It is a valid JSON encoding of the type buf.registry.policy.v1beta1.PolicyConfig.
+func MarshalPolicyConfigAsJSON(policyConfig PolicyConfig) ([]byte, error) {
+	return marshalPolicyConfigAsJSON(policyConfig)
+}
+
+/// *** PRIVATE ***
+
+// marshalPolicyConfigAsJSON marshals the given PolicyConfig to a stable JSON representation.
+//
+// This is used for the O1 digest and should not be used for other purposes.
```

Comment:
May be this part of the comment is not necessary? We're exporting it precisely because we'll use it in the BSR